File: src/test/java/com/roscopeco/jasm/model/LongMathTests.java
package com.roscopeco.jasm.model;

public interface LongMathTests {
    double testL2d(final long in);
    float testL2f(final long in);
    int testL2i(final long in);

    long testLadd(final long d1, final long d2);
    long testLand(final long a, final long b);
    int testLcmp(final long a, final long b);
    long testLdiv(final long a, final long b);
    long testLmul(final long a, final long b);
    long testLneg(final long a);
    long testLor(final long a, final long b);
    long testLrem(final long a, final long b);
    long testLshl(final long a, final int b);
    long testLshr(final long a, final int b);
    long testLsub(final long a, final long b);
    long testLushr(final long a, final int b);
    long testLxor(final long a, final long b);
}

File: src/test/kotlin/com/roscopeco/jasm/model/LongMathTests.kt
package com.roscopeco.jasm.model

interface LongMathTests {
    fun testL2d(input: Long): Double
    fun testL2f(input: Long): Float
    fun testL2i(input: Long): Int
    fun testLadd(d1: Long, d2: Long): Long
    fun testLand(a: Long, b: Long): Long
    fun testLcmp(a: Long, b: Long): Int
    fun testLdiv(a: Long, b: Long): Long
    fun testLmul(a: Long, b: Long): Long
    fun testLneg(a: Long): Long
    fun testLor(a: Long, b: Long): Long
    fun testLrem(a: Long, b: Long): Long
    fun testLshl(a: Long, b: Int): Long
    fun testLshr(a: Long, b: Int): Long
    fun testLsub(a: Long, b: Long): Long
    fun testLushr(a: Long, b: Int): Long
    fun testLxor(a: Long, b: Long): Long
}